At this time the tests for ntpd require features in the loader that are not
available in Solaris. We can run the ntp_restrict test by supplying stubs
for the required symbols. The scanner test is more complex.

--- tests/ntpd/ntp_restrict.c
+++ tests/ntpd/ntp_restrict.c
@@ -7,6 +7,12 @@
 
 /* Helper functions */
 
+unsigned long current_time;
+void 
+mon_start(int);
+void
+mon_stop(int);
+
 sockaddr_u
 create_sockaddr_u(short sin_family, unsigned short sin_port, char* ip_addr) {
 	sockaddr_u sockaddr;
--- sntp/m4/ntp_problemtests.m4.1	2017-03-30 23:02:55.085525711 +0300
+++ sntp/m4/ntp_problemtests.m4	2017-03-30 23:03:59.864424778 +0300
@@ -30,7 +30,6 @@
 case "$ntp_ept:$cross:$host" in
  no:0:*-*-freebsd6.4) ;;
  no:0:*-*-hpux11.23*) ;;
- no:0:*-*-solaris*) ;;
  *) ntp_test_ntp_restrict="yes" ;;
 esac
 AC_MSG_RESULT([$ntp_test_ntp_restrict])
